### PR TITLE
Bugfix/router and console errors

### PR DESCRIPTION
**Pull request by emrysal**
## What does this PR do?

* Fixes browser back button behaviour
* Fixes loading must be a boolean console error
* Fixes duplicate keys;

```javascript
// following removed now but just to show why the key error was thrown:
{index < queryIntegrations.data.items.length - 1 && (
  <div /* this required a key={} parameter */ className="h-[1px] w-full border-b border-gray-200" />
)}
```

### CODE REVIEW

File: apps/web/components/getting-started/components/CalendarItem.tsx [hunk 1]
```diff
@@ -20,8 +21,9 @@ const CalendarItem = (props: ICalendarItem) => {
       <InstallAppButtonWithoutPlanCheck
         type={type}
         render={(buttonProps) => (
-          <button
```
Comment (emrysal) [inline apps/web/components/getting-started/components/CalendarItem.tsx:23]: Render button can't be a native button due to reliance on Button-only parameters later down the line (isMutation in some apps)

File: apps/web/components/getting-started/steps-views/ConnectCalendars.tsx [hunk 2]
```diff
@@ -60,21 +60,17 @@ const ConnectedCalendars = (props: IConnectCalendarsProps) => {
       {firstCalendar === undefined && queryIntegrations.data && queryIntegrations.data.items.length > 0 && (
         <List className="rounded-md border border-gray-200 bg-white p-0 dark:bg-black">
           {queryIntegrations.data &&
-            queryIntegrations.data.items.map((item, index) => (
-              <>
+            queryIntegrations.data.items.map((item, i) => (
+              <div className="border-b border-gray-200 last:border-b-0" key={item.title}>
```
Comment (emrysal) [inline apps/web/components/getting-started/steps-views/ConnectCalendars.tsx:64]: ❤️ `last:`

File: apps/web/components/getting-started/steps-views/ConnectCalendars.tsx [hunk 3]
```diff
@@ -58,42 +58,35 @@ const ConnectedCalendars = (props: IConnectCalendarsProps) => {
...
             return (
-              <>
-                <div className="flex flex-row justify-center py-6" key={index}>
-                  <SkeletonAvatar width="8" height="8" className="mx-6 px-4" />
-                  <SkeletonText width="full" height="5" className="ml-1 mr-4 mt-3" />
-                  <SkeletonButton height="8" width="20" className="mr-6 rounded-md p-5" />
-                </div>
-                {index < 4 && <div className="h-[1px] w-full border-b border-gray-200" />}
-              </>
+              <li className="lasder-b-0 flex w-full flex-row justify-center py-6" key={index}>
```
Comment (alannnc) [inline apps/web/components/getting-started/steps-views/ConnectCalendars.tsx:82]:
```suggestion
              <li className="border-b-0 flex w-full flex-row justify-center py-6" key={index}>
```

Comment (alannnc) [reply apps/web/components/getting-started/steps-views/ConnectCalendars.tsx:82]: @emrysal Its border right for this one, right?

**Comment by alannnc** [comment]
Thanks for the improvements there @emrysal

**Pull request merged**